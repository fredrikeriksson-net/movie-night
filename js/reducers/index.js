import { combineReducers } from 'redux';
import movieReducer from './movie_reducer'
import tvReducer from './tv_reducer'
import movieDetailsReducer from './movie_details_reducer'
import tvDetailsReducer from './tv_details_reducer'


const rootReducer = combineReducers({
  movies: movieReducer,
  tv: tvReducer,
  movieDetails: movieDetailsReducer,
  tvDetails: tvDetailsReducer,
});

export default rootReducer;