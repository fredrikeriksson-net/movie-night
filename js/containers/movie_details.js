// React
import React, { Component, PropTypes } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchMovieDetails } from "../actions/index.js";


// Containers/Components
import MovieOverviewPopular from '../containers/movie_overview_popular.js'
import MovieOverviewUpcoming from '../containers/movie_overview_upcoming.js'
import MovieOverviewPlaying from '../containers/movie_overview_playing.js'
import MovieOverviewToprated from '../containers/movie_overview_toprated.js'

// React Router
import {Router, Route, Link, hashHistory } from 'react-router';


export class MovieDetails extends Component {
	

	componentWillMount() {
		if(this.props.fetchMovieDetails) {
			this.props.fetchMovieDetails(this.props.params.id);
		}
	}

	renderMovieDetails(){

			
			if(this.props.movieDetails[0]){
				let i = 0;
				let j = 0;
				let k = 0;
				let releaseDate = this.props.movieDetails[0].release_date;
				let year = this.props.movieDetails[0].release_date.substring(0,4);

				let genres = "";
				let productionCompanies = "";
				let productionCountries = "";
				

				let genreArr =  this.props.movieDetails[0].genres;
				let productionCompaniesArr = this.props.movieDetails[0].production_companies;
				let productionCountriesArr = this.props.movieDetails[0].production_countries;

				let genre = genreArr.forEach((genre) => {
					i !== genreArr.length - 1 ? genres += (genre.name + ", ") : genres += genre.name;
					i++;
				}) 

				let company = productionCompaniesArr.forEach((company) => {
					j !== productionCompaniesArr.length - 1 ? productionCompanies += (company.name + ", ") : productionCompanies += company.name;
					j++;
				}) 

				let country = productionCountriesArr.forEach((country) => {
					k !== productionCountriesArr.length - 1 ? productionCountries += (country.name + ", ") : productionCountries += country.name;
					k++;
				}) 
			return(
			<div id="detail-wrapper">
				<div id="detail-container">
					
					<figure>
						<img src={"https://image.tmdb.org/t/p/w500/" + this.props.movieDetails[0].poster_path}  alt="poster"/>
					</figure>
					<section>
						<h1>{this.props.movieDetails[0].title + " (" + year + ")"}</h1>
						<span>{genres}</span>
						<h2>Overview</h2>
						<p>
							{this.props.movieDetails[0].overview}
						</p>
						<h2>Facts for geeks</h2>
						<p className="facts">Release Date: {releaseDate}</p>
						<p className="facts">Countries: {productionCountries}</p>
						<p className="facts">Production Companies: {productionCompanies}</p>
						<p className="facts">Vote Avg: {this.props.movieDetails[0].vote_average}</p>
						<p className="facts">Vote Count: {this.props.movieDetails[0].vote_count}</p>
						<Link href="http://www.facebook.com">Facebook</Link>
					</section>
				</div>	
			</div>
			)
		}
		else {
			<div>Loading movies...</div>
		}
	}

	render(){
		return (
			<div className="fullPage">
				<div className="movieContainer">
					{this.renderMovieDetails()}
				</div>
			</div>
		)
	}

}


function mapStateToProps(state){
	return { movieDetails: state.movieDetails };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchMovieDetails}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieDetails);