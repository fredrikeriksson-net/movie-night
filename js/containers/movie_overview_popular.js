// React
import React, { Component } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchPopularMovies } from "../actions/index.js";

// Containers/Components
import SearchBar from "./search_bar"

// React Router
import {Router, Route, Link, hashHistory } from 'react-router';


export class MovieOverviewPopular extends Component {

	componentWillMount() {
		if(this.props.fetchPopularMovies) {
			this.props.fetchPopularMovies();
		}
		
	}
	

	renderMovies(movieList){
	
			return(
			
			<article key={Math.random()} className="movieThumb">
			
				<Link to={`/movieDetails/${movieList.id}`}>{movieList.backdrop_path === null ? <img src="img/no_image.png"></img> : <img src={"https://image.tmdb.org/t/p/w500_and_h281_bestv2/" + movieList.backdrop_path}></img>}</Link>

				<section className="info">
					
					<section className="topInfo">
						<Link to={`/movieDetails/${movieList.id}`}><h2>{movieList.title}</h2></Link>
						<h3>{movieList.release_date}</h3>
					</section>
					
					<section className="bottomInfo">
						<h3><font color="#cc0000">&#x2764;</font> {movieList.vote_count}</h3>
						<h3>Average vote: {movieList.vote_average}</h3>
					</section>
	
				</section>
			</article>
	
			)
	}

	render(){

		return (
			<div>
			<SearchBar />
				<div className="fullPage centeringDiv">
				
					<h1>Popular Movies</h1>
					
					<div className="movieContainer">
						{this.props.movies.map(this.renderMovies)}
					</div>
		
				</div>
			</div>
		)
	}

}


function mapStateToProps(state){
	return { movies: state.movies };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchPopularMovies}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieOverviewPopular);