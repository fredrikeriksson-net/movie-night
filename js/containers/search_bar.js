// React
import React, { Component } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchSearchResults } from "../actions/index.js";


// React Router
import {Router, Route, Link, hashHistory } from 'react-router';

// let category = this.props.params.menuCategory

export class SearchBar extends Component {

	onInputChange(term) {
		
		let heading = document.getElementsByTagName("h1")[0];

		this.props.fetchSearchResults(term);
		
		if(term.length === 0) {
			location.reload();//Fullösning!!! Ska hitta på något bättre i mån av tid :)
		}
		else {
			heading.innerHTML = "Search Results"
		}
	}
	
	render(){
		return (
			<div id="search-bar-wrapper">
				<input 
					type="text" 
					placeholder="Search Movies"
					onChange={e => this.onInputChange(e.target.value)} 
					/>
			</div>
		)
	}
}


function mapStateToProps(state){
	return { movies: state.movies };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchSearchResults}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);



