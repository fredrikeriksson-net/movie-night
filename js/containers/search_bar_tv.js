// React
import React, { Component } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchSearchResultsTv } from "../actions/index.js";


// React Router
import {Router, Route, Link, hashHistory } from 'react-router';

// let category = this.props.params.menuCategory

export class SearchBarTv extends Component {

	
	onInputChange(term) {
		
		let heading = document.getElementsByTagName("h1")[0];

		this.props.fetchSearchResultsTv(term);
		
		if(term.length === 0) {
			location.reload();//Fullösning!!! Ska hitta på något bättre i mån av tid :)
		}
		else {
			heading.innerHTML = "Search Results"
		}
	}
	
	render(){
		return (
			<div id="search-bar-wrapper">
				<input 
					type="text" 
					placeholder="Search Tv"
					onChange={e => this.onInputChange(e.target.value)} 
					/>
				
			</div>
		)
	}
}


function mapStateToProps(state){
	return { tv: state.tv };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchSearchResultsTv}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchBarTv);



