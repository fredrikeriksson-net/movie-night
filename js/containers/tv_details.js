// React
import React, { Component } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchTvDetails } from "../actions/index.js";


// Containers/Components
import TvOverviewPopular from '../containers/tv_overview_popular.js'
import TvOverviewUpcoming from '../containers/tv_overview_upcoming.js'
import TvOverviewPlaying from '../containers/tv_overview_playing.js'
import TvOverviewToprated from '../containers/tv_overview_toprated.js'

// React Router
import {Router, Route, Link, hashHistory } from 'react-router';


export class TvDetails extends Component {
	
	componentWillMount() {
		if(this.props.fetchTvDetails) {
			this.props.fetchTvDetails(this.props.params.id);
		}
	}

	renderTvDetails(){
		
		if(this.props.tvDetails[0]){
			let i = 0;
			let j = 0;
			let k = 0;
			let releaseDate = this.props.tvDetails[0].first_air_date;
			let year = this.props.tvDetails[0].first_air_date.substring(0,4);

			let genres = "";
			let productionCompanies = "";
			let productionCountries = "";
			

			let genreArr =  this.props.tvDetails[0].genres;
			let productionCompaniesArr = this.props.tvDetails[0].production_companies;
			
			let genre = genreArr.forEach((genre) => {
				i !== genreArr.length - 1 ? genres += (genre.name + ", ") : genres += genre.name;
				i++;
			}) 

			let company = productionCompaniesArr.forEach((company) => {
				j !== productionCompaniesArr.length - 1 ? productionCompanies += (company.name + ", ") : productionCompanies += company.name;
				j++;
			}) 
	
		return(
		<div id="detail-wrapper">
			<div id="detail-container">
				{/*<Link to={"/"}><img src="../img/arrow_back.png" /></Link>*/}
				<figure>
					<img src={"https://image.tmdb.org/t/p/w500/" + this.props.tvDetails[0].poster_path}  alt="poster"/>
				</figure>
				<section>
					<h1>{this.props.tvDetails[0].name  + " (" + year + ")"}</h1>
					<span>{genres}</span>
					<h2>Overview</h2>
					<p>
						{this.props.tvDetails[0].overview}
					</p>
					<h2>Facts for geeks</h2>
					<p className="facts">Air Date: {releaseDate}</p>
					<p className="facts">Seasons: {this.props.tvDetails[0].number_of_seasons}</p>
					<p className="facts">Production Companies: {productionCompanies}</p>
					<p className="facts">Vote Avg: {this.props.tvDetails[0].vote_average}</p>
					<p className="facts">Vote Count: {this.props.tvDetails[0].vote_count}</p>
				</section>
			</div>	
		</div>
		)
	}
	else {
		<div>Loading movies...</div>
	}
}

	render(){

		 if(this.props.tvDetails[0]){
			console.log(this.props.tvDetails[0].title)
		} 
	
		return (
			<div className="fullPage">
				<div className="movieContainer">
					{this.renderTvDetails()}
				</div>
			</div>
		)
	}

}


function mapStateToProps(state){
	return { tvDetails: state.tvDetails };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchTvDetails}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(TvDetails);