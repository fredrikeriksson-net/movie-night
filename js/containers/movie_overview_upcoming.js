// React
import React, { Component } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchUpcomingMovies } from "../actions/index.js";

// Containers/Components
import { MovieDetails } from "./movie_details.js"
import SearchBar from "./search_bar"

// React Router
import {Router, Route, Link, hashHistory } from 'react-router';


export class MovieOverviewUpcoming extends Component {

	componentWillMount() {
		if (this.props.fetchUpcomingMovies) {
			this.props.fetchUpcomingMovies();
		}
		
	}
	

	renderMovies(movieList){
	
			return(
			
			<article key={Math.random()} className="movieThumb">
			
				<Link to={`/movieDetails/${movieList.id}`}>{movieList.backdrop_path === null ? <img src="img/no_image.png"></img> : <img src={"https://image.tmdb.org/t/p/w500_and_h281_bestv2/" + movieList.backdrop_path}></img>}</Link>

				<section className="info">
					
					<section className="topInfo">
						<Link to={`/movieDetails/${movieList.id}`}><h2>{movieList.title}</h2></Link>
						<h3>{movieList.release_date}</h3>
					</section>
					
					<section className="bottomInfo">
						<h3><font color="#cc0000">&#x2764;</font> {movieList.vote_count}</h3>
						<h3>Average vote: {movieList.vote_average}</h3>
					</section>
	
				</section>
			</article>
	
			)
	}

	render(){

		return (
			<div>
			<SearchBar />
				<div className="fullPage centeringDiv">
					<h1>Upcoming Movies this month</h1>
					
					<div className="movieContainer">
						{this.props.movies.map(this.renderMovies)}
					</div>
		
				</div>
			</div>
		)
	}

}


function mapStateToProps(state){
	return { movies: state.movies };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchUpcomingMovies}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieOverviewUpcoming);