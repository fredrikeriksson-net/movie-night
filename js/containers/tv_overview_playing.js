// React
import React, { Component } from "react";

// Redux
import { connect } from "react-redux";
import { bindActionCreators } from "redux";

// Actions
import { fetchPlayingTV } from "../actions/index.js";

// Containers/Components
import { TvDetails } from "./tv_details.js"
import SearchBarTv from "./search_bar_tv"

// React Router
import {Router, Route, Link, hashHistory } from 'react-router';

// Other
//import moment from "moment"

export class TvOverviewPlaying extends Component {
	
	componentWillMount() {
		if (this.props.fetchPlayingTV) {
			this.props.fetchPlayingTV();
		}
		
	}

	renderTv(tvList){
	
			return(
			
			<article key={Math.random()} className="movieThumb">
			
				<Link to={`/TvDetails/${tvList.id}`}>{tvList.backdrop_path === null ? <img src="img/no_image.png"></img> : <img src={"https://image.tmdb.org/t/p/w500_and_h281_bestv2/" + tvList.backdrop_path}></img>}</Link>

				<section className="info">
					
					<section className="topInfo">
						<Link to={`/TvDetails/${tvList.id}`}><h2>{tvList.name}</h2></Link>
						<h3>{tvList.first_air_date}</h3>
					</section>
					
					<section className="bottomInfo">
						<h3><font color="#cc0000">&#x2764;</font> {tvList.vote_count}</h3>
						<h3><font color="#000">&#x271A;</font> {tvList.vote_average}</h3>
					</section>
	
				</section>
			</article>
	
			)
	}

	render(){

		return (
			<div>
			<SearchBarTv />
				<div className="fullPage centeringDiv">
					<h1>TV series Airing today</h1>
					
					<div className="movieContainer">
						{this.props.tv.map(this.renderTv)}
					</div>
		
				</div>
			</div>
		)
	}

}


function mapStateToProps(state){
	return { tv: state.tv };
}

function mapDispatchToProps(dispatch){
	return bindActionCreators ({fetchPlayingTV}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(TvOverviewPlaying);

// 