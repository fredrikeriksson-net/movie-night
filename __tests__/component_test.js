// React
import React from 'react';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-addons-test-utils';

// Immutable
import { fromJS } from 'immutable';

// Ensyme
import { shallow, render, mount } from 'enzyme';

//Redux
import { Provider } from 'react-redux';
import { createStore } from 'redux' 

// Components
import { MovieDetails } from '../js/containers/movie_details';
import { MovieOverviewPlaying } from '../js/containers/movie_overview_playing';
import { MovieOverviewPopular } from '../js/containers/movie_overview_popular';
import { MovieOverviewToprated } from '../js/containers/movie_overview_toprated';
import { MovieOverviewUpcoming } from '../js/containers/movie_overview_upcoming';
import { SearchBar } from '../js/containers/search_bar';
import { SearchBarTv } from '../js/containers/search_bar_tv';
import { TvDetails } from '../js/containers/tv_details';
import { TvOverviewPlaying } from '../js/containers/tv_overview_playing';
import { TvOverviewPopular } from '../js/containers/tv_overview_popular';
import { TvOverviewToprated } from '../js/containers/tv_overview_toprated';
import { TvOverviewUpcoming } from '../js/containers/tv_overview_upcoming';

// Reducers
import reducers from '../js/reducers';


describe('Components', function() {

  // MOVIE ----------------------------
	
  describe('Searchbar Movies', function() {
    	it('should render correctly', function() {
    		const tree = renderer.create(<SearchBar movies={fromJS({})} />).toJSON();
      		expect(tree).toMatchSnapshot();
    	});

    	it('should call onchange event and call fetchSearchResults', () => {
    		 const props = {
		       fetchSearchResults: (term) => { return term}
		    };

		    spyOn(props, 'fetchSearchResults');
		    props.fetchSearchResults('')

    		const wrapper = shallow(<SearchBar {...props}/>); 
    		wrapper.instance().onInputChange('')

    		expect(props.fetchSearchResults).toHaveBeenCalled()
       	});
    });


    describe('Movie Details', function() {
    	it('should render correctly', function() {
    	const tree = renderer.create(
          <Provider store={createStore(reducers)}>
    			 <MovieDetails movieDetails={fromJS({})} />
    			</Provider>).toJSON();
      		expect(tree).toMatchSnapshot();
    	});
    	it('checks if renderMovieDetails has been called', () => {
		    spyOn(MovieDetails.prototype, 'renderMovieDetails');

		    const wrapper = mount(<MovieDetails movieDetails={fromJS({})} />);

		    expect(wrapper).toBeDefined();
		    expect(MovieDetails.prototype.renderMovieDetails).toHaveBeenCalledTimes(1);
		});

    });


    describe('Movies Playing', function() {
    	it('should render correctly', function() {
    		
    		const tree = renderer.create(
          <Provider store={createStore(reducers)}>
    			 <MovieOverviewPlaying movies={fromJS({})}/>
    			</Provider>).toJSON();
      		expect(tree).toMatchSnapshot();
      	});
      	it('checks if renderMovies has been called', () => {
		    spyOn(MovieOverviewPlaying.prototype, 'renderMovies');

		    const wrapper = mount(
		    			<Provider store={createStore(reducers)}>
		    			<MovieOverviewPlaying movies={fromJS({})} />
		    			</Provider>);

		    expect(wrapper).toBeDefined();
		    expect(MovieOverviewPlaying.prototype.renderMovies).toHaveBeenCalled;
		});
    });


    describe('Movies Popular', function() {
    	it('should render correctly', function() {
    		const tree = renderer.create(
          <Provider store={createStore(reducers)}>
    			 <MovieOverviewPopular movies={fromJS({})}/>
    			</Provider>).toJSON();
      		expect(tree).toMatchSnapshot();
    	});
    	it('checks if renderMovies has been called', () => {
		    spyOn(MovieOverviewPopular.prototype, 'renderMovies');

		    const wrapper = mount(
		    			<Provider store={createStore(reducers)}>
		    			<MovieOverviewPopular movies={fromJS({})} />
		    			</Provider>);

		    expect(wrapper).toBeDefined();
		    expect(MovieOverviewPopular.prototype.renderMovies).toHaveBeenCalled;
		});
    });


    describe('Movies Toprated', function() {
    	it('should render correctly', function() {
    		const tree = renderer.create(
          <Provider store={createStore(reducers)}>
    			 <MovieOverviewToprated movies={fromJS({})}/>
    			</Provider>).toJSON();
      		expect(tree).toMatchSnapshot();
    	});
    	it('checks if renderMovies has been called', () => {
		    spyOn(MovieOverviewToprated.prototype, 'renderMovies');

		    const wrapper = mount(
		    			<Provider store={createStore(reducers)}>
		    			<MovieOverviewToprated movies={fromJS({})} />
		    			</Provider>);

		    expect(wrapper).toBeDefined();
		    expect(MovieOverviewToprated.prototype.renderMovies).toHaveBeenCalled;
		});
    });


    describe('Movies Upcoming', function() {
    	it('should render correctly', function() {
    		const tree = renderer.create(
          <Provider store={createStore(reducers)}>
    			 <MovieOverviewUpcoming movies={fromJS({})}/>
    			</Provider>).toJSON();
      		expect(tree).toMatchSnapshot();
    	});
    	it('checks if renderMovies has been called', () => {
		    spyOn(MovieOverviewUpcoming.prototype, 'renderMovies');

		    const wrapper = mount(
		    			<Provider store={createStore(reducers)}>
		    			<MovieOverviewUpcoming movies={fromJS({})} />
		    			</Provider>);

		    expect(wrapper).toBeDefined();
		    expect(MovieOverviewUpcoming.prototype.renderMovies).toHaveBeenCalled;
		});

    });


    // TV ----------------------------

    describe('Searchbar Tv', function() {
    	it('should render correctly', function() {
    		const tree = renderer.create(<SearchBarTv tv={fromJS({})} />).toJSON();
      		expect(tree).toMatchSnapshot();
    	});

    	it('should call onchange event and call fetchSearchResults', () => {
    		 const props = {
		       fetchSearchResultsTv: (term) => { return term}
		    };

		    spyOn(props, 'fetchSearchResultsTv');
		    props.fetchSearchResultsTv('')

    		const wrapper = shallow(<SearchBarTv {...props}/>); 
    		wrapper.instance().onInputChange('')

    		expect(props.fetchSearchResultsTv).toHaveBeenCalled()
       	});
    });

    describe('Tv Details', function() {
    	it('should render correctly', function() {
    		const tree = renderer.create(
          <Provider store={createStore(reducers)}>
    			 <TvDetails tvDetails={fromJS({})}/>
    			</Provider>).toJSON();
      		expect(tree).toMatchSnapshot();
    	});
    });


    describe('Tv Playing', function() {
    	it('should render correctly', function() {
    		const tree = renderer.create(
          <Provider store={createStore(reducers)}>
    			 <TvOverviewPlaying tv={fromJS({})}/>
    			</Provider>).toJSON();
      		expect(tree).toMatchSnapshot();
    	});
    	it('checks if renderTv has been called', () => {
		    spyOn(TvOverviewPlaying.prototype, 'renderTv');

		    const wrapper = mount(
		    			<Provider store={createStore(reducers)}>
		    			<TvOverviewPlaying tv={fromJS({})} />
		    			</Provider>);

		    expect(wrapper).toBeDefined();
		    expect(TvOverviewPlaying.prototype.renderTv).toHaveBeenCalled;
		});
    });


    describe('Tv Popular', function() {
    	it('should render correctly', function() {
    		const tree = renderer.create(
          <Provider store={createStore(reducers)}>
    			 <TvOverviewPopular tv={fromJS({})}/>
    			</Provider>).toJSON();
      		expect(tree).toMatchSnapshot();
    	});
    	it('checks if renderTv has been called', () => {
		    spyOn(TvOverviewPopular.prototype, 'renderTv');

		    const wrapper = mount(
		    			<Provider store={createStore(reducers)}>
		    			<TvOverviewPopular tv={fromJS({})} />
		    			</Provider>);

		    expect(wrapper).toBeDefined();
		    expect(TvOverviewPopular.prototype.renderTv).toHaveBeenCalled;
		});
    });


    describe('Tv Toprated', function() {
     	it('should render correctly', function() {
    		const tree = renderer.create(
          <Provider store={createStore(reducers)}>
					 <TvOverviewToprated tv={fromJS({})}/>
    			</Provider>).toJSON();
      		expect(tree).toMatchSnapshot();
    	});
    	it('checks if renderTv has been called', () => {
		    spyOn(TvOverviewToprated.prototype, 'renderTv');

		    const wrapper = mount(
		    			<Provider store={createStore(reducers)}>
		    			<TvOverviewToprated tv={fromJS({})} />
		    			</Provider>);

		    expect(wrapper).toBeDefined();
		    expect(TvOverviewToprated.prototype.renderTv).toHaveBeenCalled;
		});
    });


    describe('Tv Upcoming', function() {
    	it('should render correctly', function() {
    		const tree = renderer.create(
          <Provider store={createStore(reducers)}>
    			 <TvOverviewUpcoming tv={fromJS({})}/>
    			</Provider>).toJSON();
      		expect(tree).toMatchSnapshot();
    	});
    	it('checks if renderTv has been called', () => {
		    spyOn(TvOverviewUpcoming.prototype, 'renderTv');

		    const wrapper = mount(
		    			<Provider store={createStore(reducers)}>
		    			<TvOverviewUpcoming tv={fromJS({})} />
		    			</Provider>);

		    expect(wrapper).toBeDefined();
		    expect(TvOverviewUpcoming.prototype.renderTv).toHaveBeenCalled;
		});
    });

});

    
