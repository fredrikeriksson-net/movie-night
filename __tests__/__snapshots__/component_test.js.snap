exports[`Components Movie Details should render correctly 1`] = `
<div
  className="fullPage">
  <div
    className="movieContainer" />
</div>
`;

exports[`Components Movies Playing should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Movies"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Movies Now Playing
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Movies Popular should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Movies"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Popular Movies
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Movies Toprated should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Movies"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Top Rated Movies
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Movies Upcoming should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Movies"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Upcoming Movies this month
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Searchbar Movies should render correctly 1`] = `
<div
  id="search-bar-wrapper">
  <input
    onChange={[Function]}
    placeholder="Search Movies"
    type="text" />
</div>
`;

exports[`Components Searchbar Tv should render correctly 1`] = `
<div
  id="search-bar-wrapper">
  <input
    onChange={[Function]}
    placeholder="Search Tv"
    type="text" />
</div>
`;

exports[`Components Tv Details should render correctly 1`] = `
<div
  className="fullPage">
  <div
    className="movieContainer" />
</div>
`;

exports[`Components Tv Playing should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Tv"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      TV series Airing today
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Tv Popular should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Tv"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Popular TV series
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Tv Toprated should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Tv"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Top Rated TV series
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;

exports[`Components Tv Upcoming should render correctly 1`] = `
<div>
  <div
    id="search-bar-wrapper">
    <input
      onChange={[Function]}
      placeholder="Search Tv"
      type="text" />
  </div>
  <div
    className="fullPage centeringDiv">
    <h1>
      Upcoming TV series
    </h1>
    <div
      className="movieContainer" />
  </div>
</div>
`;
